### ISSUE TITLE
Comparison of 0 compile error on init.cpp

### ISSUE THREAD

**Issue by microguy** [goldcoin/Goldcoin-GLD#84]
There is an invalid argument on line 1059 of init.cpp that causes a warning on compile.

https://github.com/goldcoin/goldcoin/blob/goldcoin-master/src/init.cpp

`init.cpp:1059:72: warning: comparison of constant ‘0’ with boolean expression is always false [-Wbool-compare]
     if (GetBoolArg("-rpcserialversion", DEFAULT_RPC_SERIALIZE_VERSION) < 0)`
     
This function returns a bool type. 0 casted to bool is false but the comparison of a bool < false can never be true, because bool can never be **_less_** than **_false_**.

```
if (GetBoolArg("-rpcserialversion", DEFAULT_RPC_SERIALIZE_VERSION) < 0)
        return InitError("rpcserialversion must be non-negative.");
```

Also, the second argument to GetBoolArg() is bool fDefault however the constant passed to it, the DEFAULT_RPC_SERIALIZE_VERSION is defined as static const unsigned int in [server.h](https://github.com/goldcoin/goldcoin/blob/goldcoin-master/src/rpc/server.h) not as a bool type that the GetBoolArg() expects.

Also, the first argument "-rpcserialversion" suggests that the argument to be retrieved is not a true/false (boolean) value. All of this suggests that this is a bug and a function Getarg should be used instead of GetBoolArg in this place of the code. 

Bitcoin has fixed this and now uses GetArg instead of GetBoolArg, But fixing it is quite an extensive modification to the source. To make matters worst, this is related to SegWit which we don't use. :)

Credit to [Pavel Stepanek](https://stackoverflow.com/users/20137151/pavel-stepanek) for explanation.

**Comment by Stouse49** [comment]
We could start by deleting this code.  The serialization version is never used in the code to do anything.  It is only set as a argument to many functions.
```
if (GetBoolArg("-rpcserialversion", DEFAULT_RPC_SERIALIZE_VERSION) < 0)
        return InitError("rpcserialversion must be non-negative.");
```

The next step in removal would be to delete the line that defines this: DEFAULT_RPC_SERIALIZE_VERSION.  There are other places where related constants could be deleted.

**Comment by microguy** [comment]
Thank you Stouse. I will make the necessary changes, test the build, and make a PR today to resolve this issue.

**Comment by microguy** [comment]
This issue has been resolved in pull #88
